Dockerfile

```diff
@@ -43,32 +43,51 @@ RUN apt-get update && apt-get install -y \
   xz-utils \
   zlib1g-dev
 
-# unpack s6
+# unpack s6 and remove .tar.gz
 COPY --from=s6build /tmp /tmp
 RUN s6tar=$(find /tmp -name "s6-overlay-*.tar.gz") \
-  && tar xzf $s6tar -C / 
+  && tar xzf $s6tar -C / \
+  && rm -rf $s6tar
 
-# Install octoprint
+# Install octoprint and remove .tar.gz
 RUN	curl -fsSLO --compressed --retry 3 --retry-delay 10 \
   https://github.com/OctoPrint/OctoPrint/archive/${octoprint_ref}.tar.gz \
-	&& mkdir -p /opt/octoprint \
-  && tar xzf ${octoprint_ref}.tar.gz --strip-components 1 -C /opt/octoprint --no-same-owner
+  && mkdir -p /opt/octoprint \
+  && tar xzf ${octoprint_ref}.tar.gz --strip-components 1 -C /opt/octoprint --no-same-owner \
+  && rm -rf ${octoprint_ref}.tar.gz
 
+# install without pip wheel caching
 WORKDIR /opt/octoprint
-RUN pip install .
+RUN pip install . --no-cache-dir
 RUN mkdir -p /octoprint/octoprint /octoprint/plugins
 
-# Install mjpg-streamer
+# Install mjpg-streamer and remove .tar.gz
 RUN curl -fsSLO --compressed --retry 3 --retry-delay 10 \
   https://github.com/jacksonliam/mjpg-streamer/archive/master.tar.gz \
   && mkdir /mjpg \
-  && tar xzf master.tar.gz -C /mjpg
-
+  && tar xzf master.tar.gz -C /mjpg \
+  && rm -rf ./master.tar.gz
 
 WORKDIR /mjpg/mjpg-streamer-master/mjpg-streamer-experimental
 RUN make
 RUN make install
 
+# remove build/dev packages and it's dependencies from runtime image
+RUN apt remove -y build-essential \
+  cmake \
+  curl \  
+  g++ \
+  git \
+  libjpeg-dev \
+  imagemagick \
+  fontconfig \
+  libprotobuf-dev \
+  libv4l-dev \
+  openssh-client \
+  xz-utils \
```

Comment:
I think this is required for plugin installation through the UI, will need to test though.

Reply:
As i understand the **Plugin Manager**, it pulls the tar from github with wget/curl and extracts it, but i am not sure about this...

Reply:
wget and curl use xz-utils to extract

Reply:
plugin installation works fine with this removed. :100: